contribs/gmf/src/directives/layertree.js

```diff
@@ -0,0 +1,277 @@
+
+  var map = $scope['getMapFn']();
+  goog.asserts.assertInstanceof(map, ol.Map);
+
+  /**
+   * @type {!ol.Map}
+   * @export
+   */
+  this.map = map;
+
+  var wmsUrl = $scope['getWmsUrlFn']();
+  goog.asserts.assert(goog.isString(wmsUrl));
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.wmsUrl_ = wmsUrl;
+
+  /**
+   * @type {ngeo.LayerHelper}
+   * @private
+   */
+  this.layerHelper_ = ngeoLayerHelper;
+  this.layerHelper_.init(this.map);
+
+  $scope.$watch('source', goog.bind(function(newVal, oldVal) {
+    if (newVal !== oldVal) {
+      this['tree'] = newVal;
+    }
+  }, this));
+};
+
+
+/**
+ * Create and return a layer corresponding to the ngeo.layerTree's node.
+ * Can currently only creates WMS layers (internal or external) and WMTS layers.
+ * @param {GmfThemesNode} node Layer tree node.
+ * @return {ol.layer.Layer} The OpenLayers layer or group for the node.
+ * @export
+ */
+gmf.LayertreeController.prototype.getLayer = function(node) {
+  var wmsUrl = this.wmsUrl_;
+  var layerHelper = this.layerHelper_;
+
+  var makeLayers = function(node) {
+    var i, children = node.children;
+    var layers = new ol.Collection();
+    if (goog.isDef(children)) {
+      for (i = 0; i < children.length; i++) {
+        layers.push(makeLayers(children[i]));
+      }
+      return layerHelper.createBasicGroup(layers);
+    }
+
+    if (node.type === 'WMTS') {
+      var layerWithoutSource = new ol.layer.Tile();
+      var promise = layerHelper.createWMTSLayerFromCapabilitites(node.url,
+          node.name);
+      promise.then(function(layer) {
+        if (goog.isDef(layer)) {
+          layerWithoutSource.setSource(layer.getSource());
+        }
+      });
+      return layerWithoutSource;
+
+    } else if (node.type === 'external WMS') {
+      return layerHelper.createBasicWMSLayer(node.url, node.name);
+
+    } else {
+      return layerHelper.createBasicWMSLayer(wmsUrl, node.name);
+    }
+  };
+
+  return makeLayers(node);
+};
+
+
+/**
+ * Return a 'light' class if the current resolution of the map is out of
+ * the  min/max resolution in the node.
+ * @param {GmfThemesNode} node Layer tree node.
+ * @return {?string} 'light' or null.
+ * @export
+ */
+gmf.LayertreeController.prototype.getResolutionStyle = function(node) {
+  var style;
+  var resolution = this.map.getView().getResolution();
+  var maxExtent = node['maxResolutionHint'];
+  var minExtent = node['minResolutionHint'];
+  if (goog.isDef(minExtent) && resolution < minExtent ||
+      goog.isDef(maxExtent) && resolution > maxExtent) {
+    style = 'light';
+  }
+  return style || null;
+};
+
+
+/**
+ * Toggle activation of a node by adding or removing relative(s) layer(s) on
+ * the map.
+ * @param {ngeo.LayertreeController} treeCtrl Layer tree controller, from the
+ *     current node.
+ * @export
+ */
+gmf.LayertreeController.prototype.toggleActive = function(treeCtrl) {
+  var layer = treeCtrl.scope['layer'];
```

Comment:
`treeCtrl` is the controller of the ngeo directive right ?
The `layer` is directly in the instance of the controller. `treeCtrl.layer` doesn't work ?
https://github.com/camptocamp/ngeo/blob/master/src/directives/layertree.js#L188